app/assets/javascripts/api/plots.js

```diff
@@ -21,31 +21,81 @@ export const plotIndexUrl = (queryParams) => {
   return path;
 };
 
-export const createPlot = (manifestId, plot) =>
-  fetch(Routes.manifests_plots_create_path(PROJECT_NAME, manifestId), {
+export const createPlot = (plot) =>
+  fetch(Routes.manifests_plots_create_path(PROJECT_NAME, plot.manifestId), {
     credentials: 'same-origin',
     method: 'POST',
     headers: headers('json', 'csrf'),
-    body: JSON.stringify(plot)
+    body: JSON.stringify(plotToJson(plot))
   })
     .then(checkStatus)
     .then(parseJSON)
+    .then(plotFromJson)
 
-export const destroyPlot = (manifestId, plotId) =>
-  fetch(Routes.manifests_plots_destroy_path(PROJECT_NAME, manifestId, plotId), {
+export const destroyPlot = (plot) =>
+  fetch(Routes.manifests_plots_destroy_path(PROJECT_NAME, plot.manifestId, plot.id), {
     credentials: 'same-origin',
     headers: headers('json', 'csrf'),
     method: 'DELETE'
   })
     .then(checkStatus)
     .then(parseJSON)
 
-export const updatePlot = (manifestId, plotId, plot) =>
-  fetch(Routes.manifests_plots_update_path(PROJECT_NAME, manifestId, plotId), {
+export const updatePlot = (plot) =>
+  fetch(Routes.manifests_plots_update_path(PROJECT_NAME, plot.manifestId, plot.id), {
     credentials: 'same-origin',
     method: 'PUT',
     headers: headers('json', 'csrf'),
-    body: JSON.stringify(plot)
+    body: JSON.stringify(plotToJson(plot))
   })
     .then(checkStatus)
     .then(parseJSON)
+    .then(plotFromJson)
+
+
+
```

Comment:
Can we avoid this set of transforms? It would be simpler for the API to return JSON that we can store untransformed.

Reply:
yeah, thats a good idea.